cloud-arm/src/main/resources/templates/arm-v2.ftl

```diff
@@ -102,282 +102,291 @@
       </#if>
       "staticIpRef": "[resourceId('Microsoft.Network/publicIPAddresses', parameters('gatewaystaticipname'))]",
       "ilbBackendAddressPoolName": "${stackname}bapn",
-      "secGroupName": "${stackname}secgname",
+      <#list igs as group>
+      "${group?replace('_', '')}secGroupName": "${group?replace('_', '')}${stackname}sg",
```

Comment:
This replace('_', '') is written in many places (including ARM, and AWS) and sometimes the concatenation is done also in the template like ${group?replace('_', '')}secGroupName'))]"<#if (group_index + 1) != igs?size>,</#if>

Such thing is easier to implement in java and we have View java classes which are  designed for this purpose. I have raised the CLOUD-55526 to fix all templates.